src/libraries/System.Private.CoreLib/src/System/Text/Utf8Span.Searching.cs

```diff
@@ -274,6 +275,18 @@ private unsafe bool TryFind(Utf8Span value, StringComparison comparisonType, out
             {
                 idx = compareInfo.IndexOf(thisTranscodedToUtf16, otherTranscodedToUtf16, 0, thisTranscodedToUtf16.Length, compareOptions, &matchLength, fromBeginning);
             }
+#else
+            if (fromBeginning)
+            {
+                idx = compareInfo.IndexOf(thisTranscodedToUtf16, otherTranscodedToUtf16, 0, thisTranscodedToUtf16.Length, compareOptions);
+            }
+            else
+            {
+                idx = compareInfo.LastIndexOf(thisTranscodedToUtf16, otherTranscodedToUtf16, thisTranscodedToUtf16.Length, thisTranscodedToUtf16.Length, compareOptions);
+            }
+            // TODO_UTF8STRING: matchLength is not correct here. Need to figure this out outside of CoreLib.
```

Comment:
In practice this means that anything outside of `Ordinal` or `OrdinalIgnoreCase` would be forbidden when on netcoreapp3.0 or netstandard2.0. There's no API exposed on those TFMs that would allow getting the correct match length. Then the `matchLength = otherTranscodedToUtf16.Length;` line below would be valid.

Reply:
I don't really have any expertise here, but one idea I had was if we could go through character-by-character to try to get the correct match length. I'm not sure if that would even be possible, and I know it would be slower, but it was one idea I had on how to try to support this.

Reply:
I've changed this to only support `Ordinal` or `OrdinalIgnoreCase` outside of CoreLib.